### PR TITLE
feat(stages): duration threshold for `Execution` stage

### LINKED ISSUE #6014: Improve event reporting for execution stage with large blocks
### Describe the feature

syncing holesky, execution progress appears to be "stuck" at around 42%, but it's due to multiple massive blocks in that range with ~1400 txs, so it just takes forever
https://holesky.etherscan.io/block/656652 

since we only emit an event when a batch of txs was executed, it may appear to be stuck at a certain checkpoint when it just executes massive blocks.

I'd consider this an UX/observability issue because this state can take a few minutes (>10) and a user may think it's stuck, perhaps we should yield early if the batch takes longer than a few minutes

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by shekhirin**
Fixes https://github.com/paradigmxyz/reth/issues/6014

We lack observability in `Execution` stage when a large block range is being executed without reaching any thresholds, and the node is just silent about it.

This PR introduces a new `max_duration` threshold for `Execution` stage, which controls the maximum time spent on one batch. The default value is 10 minutes.

### CODE REVIEW

**Review by mattsse — COMMENTED** [review]
I'd like a few more docs, because not immediately obvious what this does

This also does not address https://github.com/paradigmxyz/reth/issues/6014 right?

File: Cargo.toml [hunk 1]
```diff
@@ -87,6 +87,7 @@ exclude = [".github/"]
 # Like release, but with full debug symbols. Useful for e.g. `perf`.
 [profile.debug-fast]
 inherits = "release"
+opt-level = 0
```
[the remark this answers is not in the document]
Comment (Rjected) [reply Cargo.toml:90]: adding `opt-level = 0` here would mean that this is not much different than the dev profile, why remove the optimizations? I thought the point of `debug-fast` was to enable optimizations

[the remark this answers is not in the document]
Comment (shekhirin) [reply Cargo.toml]: I was building with `debug-fast` and wanted to use an actual debugger, but any `opt-level > 0` may affect the source code in a way that makes some variables unavailable for debugger. But yes, you're right that it basically makes it almost the same as `debug`, so reverted and will not introduce in a separate PR.

**Review by mattsse — CHANGES_REQUESTED** [review]
imo this is an appropriate fix for long-running batches that improves observability for ranges with very high gas usage (DOS range mainnet, inscriptions testnets)

left some questions re use of instants

this should close #6014 ?

File: crates/stages/src/stages/execution.rs [hunk 2]
```diff
@@ -150,21 +151,22 @@ impl<EF: ExecutorFactory> ExecutionStage<EF> {
                 .block_with_senders(block_number.into(), TransactionVariant::NoHash)?
                 .ok_or_else(|| ProviderError::BlockNotFound(block_number.into()))?;
 
-            fetch_block_duration += time.elapsed();
+            let execute_start = Instant::now();
+            fetch_block_duration += execute_start - block_start;
```
[the remark this answers is not in the document]
Comment (shekhirin) [reply crates/stages/src/stages/execution.rs:155]: `time.elapsed() == Instant::now() - time`. This saves us one call to `Instant::now()`.

[the remark this answers is not in the document]
Comment (shekhirin) [reply crates/stages/src/stages/execution.rs:155]: okay, I guess here we don't care about such small savings, will fix that

[the remark this answers is not in the document]
Comment (mattsse) [reply crates/stages/src/stages/execution.rs:155]: gotcha! then this makes sense, thanks!

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**